### PR TITLE
feat: replace enterprise_support import with GradeEventContextRequest…

### PR DESCRIPTION
**Pull request by kiram15**
## Description

Replaces `get_enterprise_event_context` in lms/djangoapps/grades/events.py with `GradeEventContextRequested.run_filter`.

The filter pipeline is expected to return the final event context object, so `course_grade_passed_first_time` now uses the returned context directly rather than merging enterprise-specific fields into a base context.

If the filter raises an exception, the error is logged and re-raised.

You can see it enriching the payload with the enterprise uuid correctly when fired directly 
Screenshot 2026-05-18 at 8 29 18 PM

And in the COURSE_GRADE_PASSED_FIRST_TIME event sender also has the enterprise uuid information 
Screenshot 2026-05-18 at 8 31 56 PM

Also tested with a user that doesn't have any enterprise associations, just prints enterprise uuid empty 
Screenshot 2026-05-18 at 8 33 45 PM

## Supporting information

Jira - [ENT-11563](https://2u-internal.atlassian.net/browse/ENT-11563)

## Testing instructions

https://2u-internal.atlassian.net/wiki/spaces/SOL/pages/3710353504/Enterprise+Plugin+Ticket+Runbook
openedx-filters change is already merged into master, but make sure you have the most recent version pulled. 
edx-enterprise must be checked out on pwnage101/ENT-11563 ([PR](https://github.com/openedx/edx-enterprise/pull/2543))

Then run `docker compose exec test-shell bash -c "pytest -c pytest.local.ini tests/filters/ -v"` to validate the tests run with the edx-enterprise changes. 

## Deadline

"None" if there's no rush, or provide a specific date or event (and reason) if there is one.

## Other information

Include anything else that will help reviewers and consumers understand the change.

- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
- If your [database migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) can't be rolled back easily.

### CODE REVIEW

File: lms/djangoapps/grades/events.py [hunk 1]
```diff
@@ -166,8 +166,21 @@ def course_grade_passed_first_time(user_id, course_id):
...
+    except Exception:  # pylint: disable=broad-except
+        log.exception(
+            'GradeEventContextRequested failed for user_id=%s course_id=%s; '
+            'falling back to base grade event context',
+            user_id,
+            course_id,
+        )
+    else:
+        if filtered_context is not None:
+            context.update(filtered_context)
```
Comment (pwnage101) [inline lms/djangoapps/grades/events.py]: ❌ the filter pipeline is designed to spit out a usable object with optional enrichment already applied.  The rest of the function should be refactored to just use filtered_context.
```suggestion
    else:
        log.exception(
            'GradeEventContextRequested succeeded for user_id=%s course_id=%s',
            user_id,
            course_id,
        )
```

File: lms/djangoapps/grades/events.py [hunk 2]
```diff
@@ -166,8 +166,21 @@ def course_grade_passed_first_time(user_id, course_id):
...
+        filtered_context, _, _ = GradeEventContextRequested.run_filter(
+            context=context, user_id=user_id, course_id=course_id
+        )
+    except Exception:  # pylint: disable=broad-except
+        log.exception(
+            'GradeEventContextRequested failed for user_id=%s course_id=%s; '
+            'falling back to base grade event context',
+            user_id,
+            course_id,
+        )
```
Comment (pwnage101) [inline lms/djangoapps/grades/events.py:178]: ❌ Swallowing exceptions results in behavior change, I recommend just fatally failing instead.
```suggestion
    except Exception:  # pylint: disable=broad-except
        log.exception(
            'GradeEventContextRequested failed for user_id=%s course_id=%s',
            user_id,
            course_id,
        )
        raise
```

**Comment by pwnage101** [comment]
Please test this code path in devstack, not just running unit tests.  I'll leave it up to you to determine the best way to do so, but off the top of my head, here are some options (from least to most end-to-end):

* directly call the `course_grade_passed_first_time` function with fabricated arguments.
* emit a django signal with fabricated data and let the `listen_for_course_grade_passed_first_time` handler call the `course_grade_passed_first_time` function.
* seed/create a test course in devstack, complete it, let it emit the `COURSE_GRADE_PASSED_FIRST_TIME` signal which triggers the `listen_for_course_grade_passed_first_time` handler which calls the `course_grade_passed_first_time` function.

Ultimately, devstack inherits from the default EVENT_TRACKING_BACKENDS setting which [logs tracking events to the "tracking" logger.](https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L999-L1003)   If you're lucky, that simply ends up in the output of `make lms-logs`.  If not, then you'll likely just have to figure out a minor tweak to settings to make the tracking logger direct to stdout.

Then, please update the PR description with screenshots and/or logger clippings.

File: lms/djangoapps/grades/tests/test_events.py [hunk 3]
```diff
@@ -252,3 +253,100 @@ def test_ccx_course_passing_status_updated_emitted(self):
...
+        assert str(call_kwargs['course_id']) == str(self.course.id)
+        mock_tracker.get_tracker.return_value.context.assert_called_once_with(
+            'edx.course.grade.passed.first_time',
+            expected_context,
+        )
+
+    @patch('lms.djangoapps.grades.events.GradeEventContextRequested.run_filter')
+    def test_filter_none_return_leaves_context_intact(self, mock_run_filter):
+        """
+        If run_filter returns None, context is not overwritten.
```
Comment (pwnage101) [inline lms/djangoapps/grades/tests/test_events.py]: ❌  These unit tests are a bit outdated and assume it's possible for the pipeline to return None.  That's impossible, the type hint clearly shows only dict is possible:
https://github.com/openedx/openedx-filters/blob/8db18547dd8feb5c30e15e66e75c902388b267d0/openedx_filters/learning/filters.py#L1584

Image

If None was allowed, the type would be `dict | None`

**Review by pwnage101 — CHANGES_REQUESTED** [review]
code looks good, tests just need to be updated.

File: lms/djangoapps/grades/tests/test_events.py [hunk 4]
```diff
@@ -252,3 +253,76 @@ def test_ccx_course_passing_status_updated_emitted(self):
...
+        mock_run_filter.return_value = (filtered_context, self.user.id, self.course.id)
+
+        from lms.djangoapps.grades.events import course_grade_passed_first_time
+        with (
+            patch(
+                'lms.djangoapps.grades.events.contexts.course_context_from_course_id',
+                return_value=original_context,
+            ),
+            patch('lms.djangoapps.grades.events.tracker') as mock_tracker,
+        ):
```
Comment (pwnage101) [inline lms/djangoapps/grades/tests/test_events.py]: Nit: IMO it makes the code harder to understand when you have some `patch()` commands inside a context manager, and some as a decorator on the test method.  You want to be able to read a pile of patches in one place and not be surprised when something else is patched in a different place.  Please migrate all patches to be in one place, ideally as all decorators.

**Review by pwnage101 — APPROVED** [review]
LGTM! Just one nit.
